File: programmers/java/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        int a = num;
        
        while(a != 1 && answer != 500) {
            if(a%2 == 0) {
                a = a/2;
            } else if (a %2 == 1) {
                a = (a*3) + 1;
            }
            answer++;
        }
        if (answer == 500) {
            answer = -1;
        }
            
        return answer;
    }
}

File: programmers/Python/콜라츠 추측.py
def solution(num):
    cnt = 0
    while True:
        if num == 1:
            return cnt
        cnt +=1
        if num % 2 == 0:
            num = num / 2
        else:
            num = (num*3) + 1
        if cnt == 500:
            return -1
